src/azul/service/manifest_service.py

```diff
@@ -70,15 +74,30 @@
 logger = logging.getLogger(__name__)
 
 
+class ManifestFormat(Enum):
+    COMPACT = 'compact'
+    FULL = 'full'
+    TERRA_BDBAG = 'terra.bdbag'
+
+    @property
+    def generator_class(self) -> Type['ManifestGenerator']:
+        return {
+            'compact': CompactManifestGenerator,
```

Comment:
Duplicating the strings here is exactly what enums were designed to avoid. You could just use the actual enum values.

Besides, I also don't want this enum to be smart. The service should have the authority on which class to use for which enum value. That's a more powerful pattern than hardcoding the mapping here. It allows potential service subclasses to override the mapping, for example.